### PR TITLE
Make the android injectable with an intent

### PR DESCRIPTION
**Pull request by ErebusZ**
Make the android injectable with an intent.

### CODE REVIEW

File: app/src/main/AndroidManifest.xml [hunk 1]
```diff
@@ -38,7 +38,14 @@
             android:configChanges="orientation|keyboardHidden|keyboard|screenSize|locale|layoutDirection|fontScale|screenLayout|density|uiMode"
             android:hardwareAccelerated="true"
             android:windowSoftInputMode="adjustResize"
-            />
+            android:exported="true"
+            >
+            <intent-filter>
+               <action android:name="android.intent.action.SEND" />
+               <category android:name="android.intent.category.DEFAULT" />
+               <data android:mimeType="text/*" />
```
Comment (3asm) [inline app/src/main/AndroidManifest.xml]: Can you please remove the mimetype.

File: ostorlab_insecure_flutter_app/lib/bug_rule.dart [hunk 2]
```diff
@@ -5,7 +5,7 @@ abstract class BugRule {
   late BuildContext context;
 
   /// Runs the bug rule implementation asynchronously.
-  Future<void> run() async {
+  Future<void> run(String user_input) async {
```
Comment (3asm) [inline ostorlab_insecure_flutter_app/lib/bug_rule.dart]: Please rename to input.

File: ostorlab_insecure_flutter_app/lib/bugs/biometric_none_cryptobject.dart [hunk 3]
```diff
@@ -3,18 +3,17 @@ import 'package:local_auth/local_auth.dart';
...
 class BiometricNoneCryptObject extends BugRule {
-
   static const String _tag = 'BiometricNoneCryptObject';
 
-  String get description => 'Use Biometric authentication without a crypto Object.';
+  String get description =>
+      'Use Biometric authentication without a crypto Object.';
 
-  Future<void> run() async {
+  Future<void> run(String user_input) async {
```
Comment (3asm) [inline ostorlab_insecure_flutter_app/lib/bugs/biometric_none_cryptobject.dart]: Please rename to input.

Comment (3asm) [reply ostorlab_insecure_flutter_app/lib/bugs/biometric_none_cryptobject.dart]: Applies to all.

File: ostorlab_insecure_flutter_app/lib/bugs/command_exec.dart [hunk 4]
```diff
@@ -13,16 +13,21 @@ class CommandExec extends BugRule {
...
   /// Trigger the [BugRule]
   @override
-  Future<void> run() async {
-    TextEditingController fileController = TextEditingController(text: "/sdcard/document.pdf");
-    TextEditingController domainController = TextEditingController(text: "www.ostorlab.co");
+  Future<void> run(String user_input) async {
+    TextEditingController fileController =
+        TextEditingController(text: "/sdcard/document.pdf");
+    TextEditingController domainController =
+        TextEditingController(text: "www.ostorlab.co");
```
Comment (3asm) [inline ostorlab_insecure_flutter_app/lib/bugs/command_exec.dart]: Replace the text controller with the input. You can check the iOS code.

File: ostorlab_insecure_flutter_app/pubspec.yaml [hunk 5]
```diff
@@ -36,11 +35,15 @@ dependencies:
   build_runner: any
   built_value_generator: any
 
+
   # The following adds the Cupertino Icons font to your application.
   # Use with the CupertinoIcons class for iOS style icons.
   cupertino_icons: ^1.0.2
   convert: ^3.1.1
   local_auth: ^2.1.6
+  receive_sharing_intent: ^1.4.5
```
Comment (3asm) [inline ostorlab_insecure_flutter_app/pubspec.yaml]: receive_sharint ... is not used.

File: ostorlab_insecure_flutter_app/lib/main.dart [hunk 6]
```diff
@@ -125,7 +147,7 @@ class _MyHomePageState extends State<MyHomePage> {
               mainAxisAlignment: MainAxisAlignment.center,
               children: <Widget>[
                 ElevatedButton(
-                  onPressed: _runAll,
+                  onPressed: () => _runAll(""),
```
Comment (3asm) [inline ostorlab_insecure_flutter_app/lib/main.dart]: Use dart's default arguments.

File: ostorlab_insecure_flutter_app/lib/main.dart [hunk 7]
```diff
@@ -50,18 +53,37 @@ class MyHomePage extends StatefulWidget {
...
   String _output = '';
 
+  Future<void> _initReceiveIntent() async {
+    final receivedIntent =
+        await receive_intent.ReceiveIntent.getInitialIntent();
+    final extraList = receivedIntent!.extra!.values.toList();
+    String input = "";
+    for (var i = 0; i < extraList.length; i++) {
+      input += extraList[i];
+    }
```
Comment (3asm) [inline ostorlab_insecure_flutter_app/lib/main.dart]: Why?

**Review by 3asm — CHANGES_REQUESTED** [review]

File: app/src/main/AndroidManifest.xml (hunk 1 above)
Comment (ErebusZ) [reply app/src/main/AndroidManifest.xml]: [a91d72b](https://github.com/Ostorlab/ostorlab_insecure_android_app/pull/22/commits/a91d72b4f0fa47390729212bd4c4f3fe2b828a37)

File: ostorlab_insecure_flutter_app/lib/bug_rule.dart (hunk 2 above)
Comment (ErebusZ) [reply ostorlab_insecure_flutter_app/lib/bug_rule.dart]: [2c3d87d](https://github.com/Ostorlab/ostorlab_insecure_android_app/pull/22/commits/2c3d87d9f679ffa607a88cd4cc55780e39e06ece)

File: ostorlab_insecure_flutter_app/lib/bugs/biometric_none_cryptobject.dart (hunk 3 above)
Comment (ErebusZ) [reply ostorlab_insecure_flutter_app/lib/bugs/biometric_none_cryptobject.dart]: [2c3d87d](https://github.com/Ostorlab/ostorlab_insecure_android_app/pull/22/commits/2c3d87d9f679ffa607a88cd4cc55780e39e06ece)

File: ostorlab_insecure_flutter_app/lib/bugs/command_exec.dart (hunk 4 above)
Comment (ErebusZ) [reply ostorlab_insecure_flutter_app/lib/bugs/command_exec.dart]: [c8a9079](https://github.com/Ostorlab/ostorlab_insecure_android_app/pull/22/commits/c8a90794062425fbef2446864eb4c9584a01e667)

File: ostorlab_insecure_flutter_app/pubspec.yaml (hunk 5 above)
Comment (ErebusZ) [reply ostorlab_insecure_flutter_app/pubspec.yaml]: [a91d72b](https://github.com/Ostorlab/ostorlab_insecure_android_app/pull/22/commits/a91d72b4f0fa47390729212bd4c4f3fe2b828a37)

File: ostorlab_insecure_flutter_app/lib/main.dart (hunk 6 above)
Comment (ErebusZ) [reply ostorlab_insecure_flutter_app/lib/main.dart]: [2c3d87d](https://github.com/Ostorlab/ostorlab_insecure_android_app/pull/22/commits/2c3d87d9f679ffa607a88cd4cc55780e39e06ece)

File: ostorlab_insecure_flutter_app/lib/main.dart (hunk 7 above)
Comment (ErebusZ) [reply ostorlab_insecure_flutter_app/lib/main.dart]: [2c3d87d](https://github.com/Ostorlab/ostorlab_insecure_android_app/pull/22/commits/2c3d87d9f679ffa607a88cd4cc55780e39e06ece)

**Review by 3asm — APPROVED** [review]

**Pull request merged**